itez/beneficiary/views.py

```diff
@@ -195,6 +225,25 @@ class MedicalRecordCreateView(LoginRequiredMixin, CreateView):
     form_class = MedicalRecordForm
     template_name = "beneficiary/medical_record_create.html"
 
+    def handle_upload(self, f, destination_directory=None):
+        fullpath = f"{settings.MEDIA_ROOT}/supporting_documents/{destination_directory}"
+        if not os.path.exists(fullpath):
+            os.makedirs(fullpath)
+
+        with open(f"{fullpath}/{f.name}", "wb+") as file:
+            for chunk in f.chunks():
+                file.write(chunk)
+
+    def generate_upload_dirname(self, beneficiary_name):
+        uuid_chars = str(uuid.uuid4())[:8]
+        directory_name = f"{beneficiary_name}_{uuid_chars}"
+        return directory_name
+
+    def create_files_dict(self, directory=None, filenames=[]):
+        d = {}
+        d[directory] = filenames
+        return d
```

Comment:
I would move these 3 methods to utility module as standalone generic reusable functions

Reply:
That makes sense. I will move it to the utils module.